src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -856,6 +858,17 @@ GraphPatternOperation Visitor::visitSpatialQuery(
     parseSpatialQuery(spatialQuery, op);
   }
 
+  try {
+    // We convert the spatial query to a spatial join configuration and discard
+    // its result here to detect errors early and report them to the user with
+    // highlighting. It's only a small struct so not much is wasted.
+    spatialQuery.toSpatialJoinConfiguration();
+  } catch (const parsedQuery::MagicServiceException& magicExc) {
+    reportError(ctx, magicExc.what());
+  } catch (const parsedQuery::SpatialSearchException& magicExc) {
+    reportError(ctx, magicExc.what());
+  }
```

Comment:
Can you (it is not much code), 
Do the same thing for the `PathSearch` (better local errors).

Reply:
I was about to implement this, but I found out that this is indeed not straightforward and analogous to the spatial search: The PathSearch needs the local vocabulary to figure out if its configuration is valid. But at this stage in the parsing we don't have the vocabulary yet or am I missing something?

Reply:
Yes okay, if that is more code than necessary (because of a flawed design there), then leave it as is:).